src/screens/Server.js

```diff
@@ -87,26 +125,63 @@ class Server extends React.Component {
       return;
     }
 
-    this.setState({ loading: true, errorMessage: '', selectedServer: newServer });
-    const currentServer = hathorLib.helpers.getServerURL();
-    // Update new server in local storage
-    hathorLib.wallet.changeServer(newServer);
-    hathorLib.versionApi.getVersion((data) => {
-      // If the user selected a testnet server, we show a modal to confirm the operation
-      if (data.network !== 'mainnet') {
+    this.setState({
+      loading: true,
+      errorMessage: '',
+      selectedServer: newBaseServer,
+      selectedWsServer: newWsServer,
+    });
+
+    const currentServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseUrl() : 
+      hathorLib.config.getServerUrl();
+
+    const currentWsServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseWsUrl() :
+      '';
+
+    // Update new server in storage and in the config singleton
+    this.props.wallet.changeServer(newBaseServer);
+
+    // We only have a different websocket server on the wallet-service facade, so update the config singleton
+    if (this.props.useWalletService) {
+      this.props.wallet.changeWsServer(newWsServer);
+    }
+
+    try {
+      const versionData = await this.props.wallet.getVersionData();
+
+      if (versionData.network !== 'mainnet') {
+        const network = versionData.network;
+        let selectedNetwork = network;
+
+        // Network might be 'testnet-golf' or 'testnet-charlie'
+        if (network.startsWith('testnet')) {
+          selectedNetwork = 'testnet';
+        }
+
+        this.setState({
+          selectedNetwork,
+        });
+
         // Go back to the previous server
         // If the user decides to continue with this change, we will update again
-        hathorLib.wallet.changeServer(currentServer);
+        this.props.wallet.changeServer(currentServer);
+        if (this.props.useWalletService) {
+          hathorLib.config.setWalletServiceBaseWsUrl(currentWsServer);
+        }
         $('#alertModal').modal('show');
-        this.setState({ loading: false });
       } else {
+        // We are on mainnet, so set the network on the singleton and storage
+        hathorLib.config.setNetwork('mainnet');
+        helpers.updateNetwork('mainnet');
         this.executeServerChange();
       }
-    }, () => {
+    } catch (e) {
       // Go back to the previous server
-      hathorLib.wallet.changeServer(currentServer);
+      this.props.wallet.changeServer(currentServer);
```

Comment:
You should go back to the ws server as well in case of wallet service, right?

Reply:
I should, done in 2ae8b4214d1015beba3730f90580a3a0200f2494